core/tests/widgets/test_container_debug_mode.py

```diff
@@ -0,0 +1,76 @@
+from pytest import MonkeyPatch
+from travertino.colors import color
+
+import toga
+
+
+# test that a container-like widget in normal mode has a default background
+def test_box_no_background_in_normal_mode():
+    """A Box has no default background."""
+    # Disable layout debug mode
+    with MonkeyPatch.context() as mp:
+        mp.setenv("TOGA_DEBUG_LAYOUT", "0")
+        box = toga.Box()
+        # assert that the bg is default
+        assert hasattr(box.style, "background_color")
+        assert not box.style.background_color
+
+
+# test that a non-container-like widget in layout debug mode has a default background
+def test_button_debug_background():
+    """A Button in layout debug mode has a default background."""
+    # Enable layout debug mode
+    with MonkeyPatch.context() as mp:
+        mp.setenv("TOGA_DEBUG_LAYOUT", "1")
+        button = toga.Button()
+        # assert that the bg is default
+        assert hasattr(button.style, "background_color")
+        assert not button.style.background_color
```

Comment:
As above:
* monkeypatch can be used as a fixture.
* the `hasattr` won't be asserting anything
* A positive value check would be preferable.

The same pattern applies to the subseqeunt tests as well.